### PR TITLE
Update jobs to reflect the latest validation database

### PR DESCRIPTION
**Pull request by loic-sharma**
Updates jobs to the latest validation database entities, which had breaking changes.

See: https://github.com/NuGet/ServerCommon/pull/357

Part of: https://github.com/NuGet/Engineering/issues/3576

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs [hunk 1]
```diff
@@ -72,11 +72,18 @@ public Task<bool> DoesPackageFileExistAsync(PackageValidationSet validationSet)
 
         public async Task StorePackageFileInBackupLocationAsync(PackageValidationSet validationSet, Stream packageFile)
         {
+            if (validationSet.ValidatingType == ValidatingType.Generic)
```
Comment (joelverhagen) [inline src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs:75]: I don't understand how you picked this spot to check the `ValidatingType`. To me it seems that should should be done at the very top of orchestrator or anywhere `.Value` is used.

Comment (loic-sharma) [reply src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs:75]: We do check at the very top of the orchestrator:

1. The `CheckValidator` message currently requires that validation set's type matches the Orchestrator's type. In other words, the NuGet package orchestrator will throw if it receives a `CheckValidator` message for a symbols or generic package validation set.
2. The `ProcessValidationSet` message cannot be created for a generic validation set (see [this](https://github.com/NuGet/ServerCommon/pull/357/files#diff-4c49a2687a2346a21c0e29b8ec02c2494d7497e9ed2d48d635dd188b249e3165R26-R32)).

So currently it is impossible to process a "generic" validation set using the Orchestrator.

So why did I add a check here? This is a public method that accepts a validation set but only works for NuGet packages or symbols. You should never call this method with a "generic" validation set. There are likely many more of these methods, but I don't intend to add these checks everywhere as part of this change. Once I start allowing "generic" validation sets in the Orchestrator, I will need to do a more thorough pass through all these methods and add the necessary checks.

Does that seem reasonable? Please let me know if you have any feedback!

Comment (loic-sharma) [reply src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs:75]: OK added the new tests to cover the `CheckValidator` message case here: https://github.com/NuGet/NuGet.Jobs/pull/961/commits/21ba93ade658312babaefe61621d58aab23013b0

File: Directory.Build.props [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
   <!-- NuGet dependencies shared across projects -->
   <PropertyGroup>
-    <ServerCommonPackageVersion>2.83.0</ServerCommonPackageVersion>
+    <ServerCommonPackageVersion>2.84.0-loshar-uvp-db-4454369</ServerCommonPackageVersion>
```
Comment (loic-sharma) [inline Directory.Build.props:5]: TODO

File: src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Services.Validation.Orchestrator/ValidationPackageFileService.cs]: Seems reasonable. Thanks for the clarification.

**Review by joelverhagen — APPROVED** [review]

**Review by zhhyu — APPROVED** [review]

**Pull request merged**